### PR TITLE
add buffer management, compilization, hashtable etc stuff

### PR DESCRIPTION
**Pull request by cakebytheoceanLuo**
add sections:
- buffer management: to convert In-Memory DBMSs to Disk-based DBMSs
- hash table 

add some adaptive compilazation papers and some stuff

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -126,18 +129,29 @@ Papers:
...
 - 2018, [Everything You Always Wanted to Know About Compiled and Vectorized Queries But Were Afraid to Ask](http://www.vldb.org/pvldb/vol11/p2209-kersten.pdf), VLDB
+- 2018, [Adaptive Execution of Compiled Queries](https://db.in.tum.de/~leis/papers/adaptiveexecution.pdf), ICDE
+- 2018, [Adaptive Optimization of Very Large Join Queries](https://db.in.tum.de/~radke/papers/hugejoins.pdf), SIGMOD
 
 ### Join
 
 Papers:
 
 - 2013, [Multi-Core, Main-Memory Joins: Sort vs. Hash Revisited](http://www.vldb.org/pvldb/vol7/p85-balkesen.pdf), VLDB
+- 2015, [Unnesting Arbitrary Queries](http://www.btw-2015.de/res/proceedings/Hauptband/Wiss/Neumann-Unnesting_Arbitrary_Querie.pdf), BTW
```
Comment (zz-jason) [inline README.md]: this section is focused on join implementations, it's better to put this paper to a section like `Subquery Optimization` in `Query Optimizer`

File: README.md [hunk 2]
```diff
@@ -126,18 +129,29 @@ Papers:
...
 - 2017, [Looking Ahead Makes Query Plans Robust](http://www.vldb.org/pvldb/vol10/p889-zhu.pdf), VLDB
 
 ## DDL
 
 - 2013, [Online, Asynchronous Schema Change in F1](https://research.google.com/pubs/archive/41376.pdf), VLDB
 
+### Hash Table
```
Comment (zz-jason) [inline README.md]: please put this section to `Query Execution`, instead of `DDL`.

File: README.md [hunk 3]
```diff
@@ -187,6 +201,12 @@ Papers:
 
 ## Storage
 
+### Buffer Management
+
+- 2018, [Managing Non-Volatile Memory in Database Systems](https://db.in.tum.de/people/sites/vanrenen/papers/HyMem.pdf?lang=de), SIGMOD
+- 2018, [LeanStore: In-Memory Data Management Beyond Main Memory](https://db.in.tum.de/~leis/papers/leanstore.pdf), ICDE
+- 2020, [Umbra: A Disk-Based System with In-Memory Performance](https://db.in.tum.de/~radke/papers/hugejoins.pdf), CIDR
```
Comment (zz-jason) [inline README.md]: Is the `Buffer Management` the main contribution of these papers?

File: README.md (hunk 1 above)
Comment (cakebytheoceanLuo) [reply README.md]: Ok.Done.

File: README.md (hunk 2 above)
Comment (cakebytheoceanLuo) [reply README.md]: Sorry. That was a mistake.

File: README.md (hunk 3 above)
Comment (cakebytheoceanLuo) [reply README.md]: They are current state-of-art research, aiming to introduce a advanced buffer manager in Disk-based DBMSs to have the same level performance as In-Memory DBMSs.

For your question, I added the CMU lecture link and two old papers under this topic. 
The main contribution is **the 5 minite rule** by Jim Gray, which is included.

Comment (zz-jason) [reply README.md]: got it. 👍

File: README.md [hunk 4]
```diff
@@ -149,13 +152,18 @@ Papers:
...
 - 2005, [An Improved Data Stream Summary: The Count-Min Sketch and its Applications](http://dimacs.rutgers.edu/~graham/pubs/papers/cm-full.pdf), Journal of Algorithms
 - 2007, [New Estimation Algorithms for Streaming Data: Count-min Can Do More](http://webdocs.cs.ualberta.ca/~drafiei/papers/cmm.pdf)
 - 2017, [Adaptive Statistics in Oracle 12c](http://www.vldb.org/pvldb/vol10/p1813-zait.pdf), VLDB
 
 Books:
 - [Synopses for Massive Data: Samples, Histograms, Wavelets, Sketches](https://db.cs.berkeley.edu/cs286/papers/synopses-fntdb2012.pdf)
 
+### Subquery Optimization
+
+- 2015, [Unnesting Arbitrary Queries](http://www.btw-2015.de/res/proceedings/Hauptband/Wiss/Neumann-Unnesting_Arbitrary_Querie.pdf), BTW
```
Comment (zz-jason) [inline README.md]: there is already a section for [Subquery Optimization](https://github.com/pingcap/awesome-database-learning#subquery-optimization), you can put this paper into that section.

Comment (cakebytheoceanLuo) [reply README.md]: ok

File: README.md [hunk 5]
```diff
@@ -178,6 +183,8 @@ Papers:
 - 2011, [Efficiently Compiling Efficient Query Plans for Modern Hardware](https://www.vldb.org/pvldb/vol4/p539-neumann.pdf), VLDB
 - 2017, [Relaxed Operator Fusion for In-Memory Databases: Making Compilation, Vectorization, and Prefetching Work Together At Last](http://www.vldb.org/pvldb/vol11/p1-menon.pdf), VLDB
 - 2018, [Everything You Always Wanted to Know About Compiled and Vectorized Queries But Were Afraid to Ask](http://www.vldb.org/pvldb/vol11/p2209-kersten.pdf), VLDB
+- 2018, [Adaptive Execution of Compiled Queries](https://db.in.tum.de/~leis/papers/adaptiveexecution.pdf), ICDE
+- 2018, [Adaptive Optimization of Very Large Join Queries](https://db.in.tum.de/~radke/papers/hugejoins.pdf), SIGMOD
```
Comment (zz-jason) [inline README.md]: It's better to put this paper into the `Join Order Optimization` section.

Comment (cakebytheoceanLuo) [reply README.md]: Ok. I suppost only the `Adaptive Optimization of Very Large Join Queries`.

File: README.md [hunk 6]
```diff
@@ -239,6 +253,22 @@ Papers:
...
 ## Storage
 
+### Buffer Management
+
+Courses:
+
+- CMU [Database Systems (15-445/645)](https://15445.courses.cs.cmu.edu/fall2019/schedule.html), thanks to [Andy Pavlo](http://www.cs.cmu.edu/~pavlo/)
+    - [Buffer Pools](https://15445.courses.cs.cmu.edu/fall2019/schedule.html#sep-11-2019)
+
+
```
Comment (zz-jason) [inline README.md]: one empty line is sufficient.

File: README.md [hunk 7]
```diff
@@ -24,16 +24,19 @@ Please submit a pull request if there is any material that you think should be i
     * [Functional Dependency & Physical Properties](#functional-dependency--physical-properties)
     * [Cost Model](#cost-model)
     * [Statistics](#statistics)
+    * [Subquery Optimization](#subquery-optimization)
```
Comment (zz-jason) [inline README.md]: please remove this incorrect toc.

Comment (cakebytheoceanLuo) [reply README.md]: done

File: README.md (hunk 6 above)
Comment (cakebytheoceanLuo) [reply README.md]: done

**Pull request merged**